src/models/gamma_poisson_model.py

```diff
@@ -225,7 +227,7 @@ def _expected_impressions(self, N, alpha, beta):
         Returns:
           Expected size of impression inventory for a population of N users.
         """
-        return N * alpha / beta
+        return N * (1.0 + alpha * beta)
```

Comment:
Could you elaborate how the expected impression is equal to this line?

Reply:
I think I found the source of confusion.  According to Wikipedia, the PMF of the negative binomial is as follows:

f_wikipedia(k | p, r) = C(k + r - 1, k) (1 - p)^r p^k

However, scipy.stats.nbinom defines the PMF as follows:

f_scipy(k | p, r) = C(k + r - 1) p^r (1 - p)^k

In the two definitions, p has been swapped with 1-p.  In other words, f_scipy(k | p, r) = f_wikipedia(k | 1-p, r). 

From the Wikipedia page, the mean of a negative binomial with parameters p and r is pr / (1 - p).  However, replacing 1-p with p in this definition, the mean of a negative binomial according to scipy with parameters p and r is (1 - p) r / p.

When reparametrizing a Gamma-Poisson distribution as a negative binomial, we use the parameterization r = alpha, p = 1/(1 + beta).  In other words, beta = (1 - p) / p.  Thus, plugging these values into the formula for the mean, we get that the mean of the distribution is (1 - p) r / p = alpha * beta.

In our case, we add +1 because the distribution is shifted by 1.  And we multiply by N because N is the total number of users in the universe.

In any event, it doesn't really matter which parameterization we use, as long as everything is self-consistent.  I think the unit tests and the colab demonstrate that everything in this implementation is self-consistent.

I changed the code so that instead of using the explicit formula for the mean, it uses scipy.stats.nbinom.mean.  Hopefully this will remove all remaining doubt.

https://en.wikipedia.org/wiki/Negative_binomial_distribution
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.nbinom.html

Reply:
Thank you so much for the detailed explanation! It all makes sense to me now.